### ISSUE TITLE
java.lang.NoClassDefFoundError Error

### ISSUE THREAD

**Issue by shivapbhusal** [microsoft/sql-spark-connector#31]
I downloaded the jar file and added it to my project and updated the code accordingly. 

```scala
dataFrame
      .write
      .format("com.microsoft.sqlserver.jdbc.spark")
      .mode("append")
      .option("url", url)
      .option("dbtable", destinationTable)
      .option("user", username)
      .option("password", password)
      .option("reliabilityLevel", "BEST_EFFORT")
      .option("tableLock", "true")
      .save()
```
While running the code, I get an issue below : 

```
java.lang.NoClassDefFoundError: org/apache/spark/sql/execution/datasources/jdbc/JdbcOptionsInWrite
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	at java.lang.Class.getConstructor0(Class.java:3075)
	at java.lang.Class.newInstance(Class.java:412)
	at org.apache.spark.sql.execution.datasources.DataSource.planForWriting(DataSource.scala:544)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:278)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:267)
	at td2.spark.common.SQLUtilities.writeToSqlTable(SQLUtilities.scala:78)
	at td2.spark.da.TD2DBConnector$.loadToDestination(TD2DBConnector.scala:382)
	at td2.spark.da.TD2DBConnector$.insertDASessionPAAS(TD2DBConnector.scala:171)
	at td2.spark.da.MainWorkflow$.main(MainWorkflow.scala:68)
	at td2.spark.da.MainWorkflow.main(MainWorkflow.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.spark.sql.execution.datasources.jdbc.JdbcOptionsInWrite
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 24 more

```

**Comment by cchighman** [comment]
Duplicate of #26

**Comment by cchighman** [comment]
Assuming you are using Spark < 2.4.x on a Databricks 6.x cluster...

**Comment by shivapbhusal** [comment]
Yes, that is correct.

**Comment by cchighman** [comment]
Feel free to reply back here and mention me if the comments in Issue #26 do not get you unblocked.

**Comment by shivapbhusal** [comment]
Thank you. Let me go through the comments in Issue  #26 and try the suggestions.

**Comment by shivsood** [comment]
Closing this as dupe of #26 .

**Comment by ChetanVasudevan** [comment]
@shivsood  @shivapbhusal  I still have this issue, please can someone tell me what steps you followed to resolve this. I also read issue #26 but still stuck

**Comment by LingangJiang** [comment]
i have this issue too

**Comment by SUDIPSJE** [comment]
I am having the same issue. BTW my Sparc Cluster Version is 2.3.0. Can that be causing this issue?
